src/java/org/apache/cassandra/service/reads/DataResolver.java

```diff
@@ -121,22 +121,12 @@ private boolean needsReplicaFilteringProtection()
         if (command.rowFilter().isEmpty())
             return false;
 
-        IndexMetadata indexMetadata = command.indexMetadata();
-
+        Index.QueryPlan queryPlan = command.indexQueryPlan();
+        IndexMetadata indexMetadata = queryPlan == null ? null : queryPlan.getFirst().getIndexMetadata();
```

Comment:
I think we don't need getting the `IndexMetatada` anymore, IIRC those checks where here to avoid getting the `ColumnFamilyStore` and the `Index` if possible, but now the query plan has all we need.